trieste/acquisition/function/active_learning.py

```diff
@@ -245,3 +245,153 @@ def acquisition(x: TensorType) -> TensorType:
+    ) -> None:
+        """
+        :param integration_points: set of points to integrate the prediction variance over.
+        :raise ValueError (or InvalidArgumentError): If ``integration_points`` does not have
+                exactly two dimensions, or if the first dimension has size 0.
+        """
+        self._integration_points = integration_points
+        self._threshold = threshold
+
+    def __repr__(self) -> str:
+        """"""
+        return f"IntegratedVarianceReduction(threshold={self._threshold!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return integrated_variance_reduction(model, self._integration_points, self._threshold)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class integrated_variance_reduction(AcquisitionFunctionClass):
+    """
+    The reduction of the average of the predicted variance over the integration points
+    (a.k.a. Integrated Means Square Error or IMSE criterion).
+    See :cite:`Picheny2010` for details.
+
+    If no threshold is provided, the goal is to learn a globally accurate model, and
+    the original variance is used. Otherwise, learning is 'targeted' towards regions
+    where the GP is close to particular values, and the variance is weighted by the
+    posterior GP pdf evaluated at the threshold (if a single value is given) or by the
+    probability that the GP posterior belongs to the interval between the 2 thresholds.
+    """
+
+    def __init__(
+        self,
+        model: ProbabilisticModel,
+        integration_points: TensorType,
+        threshold: Optional[Sequence[float]] = None,
+    ):
+        """
+        :param model: The model of the objective function.
+        :param integration_points: points over which to integrate the objective prediction variance
+        :param threshold: either None, or a sequence of 1 or 2 float values
+        :raise ValueError (or InvalidArgumentError): If ``threshold`` has more than 2 values.
+        """
+        if not hasattr(model, "conditional_predict_f"):
+            raise AttributeError(
+                """
+                Integrated variance reduction only supports models with a
+                conditional_predict_f method.
+                """
+            )
+
+        self._model = model
+
+        tf.debugging.assert_equal(
```

Comment:
I placed these tests here rather than in `IntegratedVarianceReduction` because it made my life slightly easier. We haven' t been very consistent in trieste about this kind of tests (). Not sure if it matters? @apaleyes @hrvoje

Reply:
yeah, we haven't been consistent about that (this design with builders and acq fncs is also inelegant to me, lots of duplication), personally I'm trying to put these in the builder as thats what is executed first, but it is fine here as well I think, as it is not executed for every call